pkg/proof/README.md

```diff
@@ -0,0 +1,198 @@
+
+### Data root
+
+The Celestia block [data root](https://celestiaorg.github.io/celestia-app/specs/data_structures.html) is computed
+by generating a binary merkle root over the set of row roots and column roots of the square:
+
+<img src="img/data_root.png" alt="data root generation" width="443"/>
+
+This allows inclusion proofs of the shares to the data root which will be discussed below.
+
+## Inclusion proof
+
+### Share to data root inclusion proof
+
+To prove that a share was included in a Celestia block,
+we can create an inclusion proof from the share to the data root.
+This proof consists of the following:
+
+#### Share to row root namespace merkle inclusion proof:
+
+First, we prove inclusion of the share in question to the row root it belongs to.
+If we take, for example, share *three*, its inclusion proof can be constructed using the inner nodes in brown: 
+
+<img src="img/share_to_row_root_proof.png" alt="share to row root proof" width="443"/>
+
+Now, if we have this inclusion proof, and we verify it against the `RowRoot1`, the proof would be valid.
+
+#### Row root to data root inclusion proof
+
+Now that we proved the share to the row root,
+we will need to complete the proof with a row root to the data root inclusion proof.
+This means that we will be proving that the row root commits to the share, and the data root commits to the row root.
+Since share *three* is committed to by `RowRoot1`, we will be proving that row root to the data root. 
+
+<img src="img/row_root_to_data_root_proof.png" alt="row root to data root proof" width="443"/>
+
+The inner nodes in brown are the nodes needed to construct the merkle inclusion proof.
+
+So, if we have access to these two proofs,
+the [share to row root inclusion proof](#share-to-row-root-namespace-merkle-inclusion-proof) and [row root to data root inclusion proof](#row-root-to-data-root-inclusion-proof),
+we can successfully prove inclusion of share *three* to the Celestia data root.
+
+### Prove share commitment inclusion to data root
+
+To prove that a blob,
+referenced by its *share commitment*, was included in a Celestia block, we need the following proofs:
+
+#### Subtree roots inclusion proof to the data root
+
+As we've seen in the [row roots generation](#row-roots) section,
+the subtree roots are inner nodes of the namespace merkle tree used to generate the row root.
+This means that it's possible to create an inclusion proof of a subtree root to the row root:
+
+<img src="img/subtree_root_to_row_root.png" alt="subtree root to row root proof" width="443"/>
+
+The inner nodes in brown would be used to construct the `SR2` node inclusion proof to the `RowRoot3`.
+
+After having this proof, we will need to prove the `RowRoot3` inclusion to the data root.
+This is described in the [row root to data root inclusion proof](#row-root-to-data-root-inclusion-proof) section.
+
+With this, we would be able to prove that `SR3` was committed to by a Celestia data root.
```

Comment:
[blocking] SR3 has never been referenced before and it's not in the diagram. I think this is incorrect.

Reply:
[c07f5fa](https://github.com/celestiaorg/celestia-app/pull/3245/commits/c07f5fac6d5ae3d9019e54a4ff87df8a0743f898)